front/lib/actions/configuration/mcp.ts

```diff
@@ -62,49 +62,45 @@ export async function fetchMCPServerActionConfigurations(
   >();
 
   for (const config of mcpServerConfigurations) {
-    const { agentConfigurationId, sId, id, serverType } = config;
+    const { agentConfigurationId, sId, id, mcpServerViewId } = config;
 
     const dataSourceConfiguration =
       dataSourceConfigurations.filter(
         (ds) => ds.mcpServerConfigurationId === config.id
       ) ?? [];
 
+    const mcpServerView = await MCPServerViewResource.fetchByModelPk(
+      auth,
+      mcpServerViewId
+    );
+
+    if (!mcpServerView) {
+      throw new Error(
+        `MCPServerView with mcpServerViewId ${mcpServerViewId} not found.`
+      );
+    }
+
+    const serverType = mcpServerView.serverType;
+
     let metadata: MCPServerMetadata | null = null;
     let remoteMCPServerId: string | null = null;
     if (serverType === "remote") {
-      if (!config.remoteMCPServerId) {
-        throw new Error(
-          `Remote MCP server ID is required for remote server type.`
-        );
-      }
-
-      const remoteMCPServer = await RemoteMCPServerResource.findByPk(
-        auth,
-        config.remoteMCPServerId
-      );
-      if (!remoteMCPServer) {
-        throw new Error(
-          `Remote MCP server with remoteMCPServerId ${config.remoteMCPServerId} not found.`
-        );
-      }
-      remoteMCPServerId = RemoteMCPServerResource.modelIdToSId({
-        id: config.remoteMCPServerId,
-        workspaceId: auth.getNonNullableWorkspace().id,
-      });
+      const remoteMCPServer = await mcpServerView.getRemoteMCPServer(auth);
```

Comment:
Ideally we would load the remoteMCPServerResource as part of loading the MCPServerViewResource so that we have mcpServerView.mcpServer always available?